plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -671,4 +674,181 @@ private function normalize_raw_meta_data( array $data ) : array {
+					'wp wc cot sync',
+				)
+			);
+			$enable_hpos = false;
+		}
+
+		if ( $assoc_args['with-sync'] && $table_exists ) {
+			if ( $data_synchronizer->data_sync_is_enabled() ) {
+				WP_CLI::warning( __( 'Sync is already enabled.', 'woocommerce' ) );
+			} else {
+				$feature_controller->change_feature_enable( DataSynchronizer::ORDERS_DATA_SYNC_ENABLED_OPTION, true );
+				WP_CLI::log( __( 'Sync enabled.', 'woocommerce' ) );
+			}
+		}
+
+		if ( ! $enable_hpos ) {
+			WP_CLI::error( __( 'HPOS could not be enabled, see the errors above', 'woocommerce' ) );
+			return;
+		}
+
+		/** CustomOrdersTableController instance @var CustomOrdersTableController $cot_status */
+		$cot_status = wc_get_container()->get( CustomOrdersTableController::class );
+		if ( $cot_status->custom_orders_table_usage_is_enabled() ) {
+			WP_CLI::warning( __( 'HPOS is already enabled.', 'woocommerce' ) );
+		} else {
+			$feature_controller->change_feature_enable( 'custom_order_tables', true );
+			WP_CLI::log( __( 'HPOS enabled.', 'woocommerce' ) );
+		}
+	}
+
+	/**
+	 * Disables custom order tables (HPOS) and posts to authoritative if HPOS and post tables are in sync.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--with-sync]
+	 * : Also disables sync (if it's currently enabled).
+	 * ---
+	 * default: false
+	 * ---
+	 *
+	 * ### EXAMPLES
+	 *
+	 *  # Disable HPOS.
+	 *  wp wc cot disable
+	 *
+	 * @param array $args Positional arguments passed to the command.
+	 * @param array $assoc_args Associative arguments (options) passed to the command.
+	 */
+	public function disable( $args, $assoc_args ) {
+		$assoc_args = wp_parse_args(
+			$assoc_args,
+			array(
+				'with-sync' => false,
+			)
+		);
+
+		WP_CLI::log( __( 'Running pre-disable checks...', 'woocommerce' ) );
+
+		/** DataSynchronizer instance @var DataSynchronizer $data_synchronizer */
+		$data_synchronizer = wc_get_container()->get( DataSynchronizer::class );
+		$pending_orders    = $data_synchronizer->get_total_pending_count();
+		if ( $pending_orders > 0 ) {
+			return WP_CLI::error(
+				sprintf(
+					// translators: %s is the command to run (wp wc cot sync).
+					__( '[Failed] There are orders pending sync. Please run `%s` to sync pending orders.', 'woocommerce' ),
+					'wp wc cot sync',
+				)
+			);
+		}
+
+		/** FeaturesController instance @var FeaturesController $feature_controller */
+		$feature_controller = wc_get_container()->get( FeaturesController::class );
+
+		/** CustomOrdersTableController instance @var CustomOrdersTableController $cot_status */
+		$cot_status = wc_get_container()->get( CustomOrdersTableController::class );
+		if ( ! $cot_status->custom_orders_table_usage_is_enabled() ) {
+			WP_CLI::warning( __( 'HPOS is already disabled.', 'woocommerce' ) );
+		} else {
+			$feature_controller->change_feature_enable( 'custom_order_tables', false );
+			WP_CLI::log( __( 'HPOS disabled.', 'woocommerce' ) );
```

Comment:
Minor/optional:

```suggestion
			WP_CLI::success( __( 'HPOS disabled.', 'woocommerce' ) );
```

Reply:
Addressed in https://github.com/woocommerce/woocommerce/pull/39865/commits/5d858e07f86129df8dec6bf4630c3545489c21f5